### PR TITLE
Fix rdkit tests

### LINKED ISSUE #4624: Unpin rdkit imports once rdkit supports numpy 2.0
With #4620 we are adding import skips when trying to use rdkit with numpy > 2.0. Once rdkit has 2.0 support we should remove those skips.

Tracking issue: https://github.com/rdkit/rdkit/issues/7477

### PR DESCRIPTION
**Pull request by IAlibay**
Fixes #4624 

Changes made in this Pull Request:
 - Update coordinate checks
 - Remove numpy >=2 skips (latest release supports numpy 2.0)

PR Checklist
------------
 - [x] Tests?
 - Docs?
 - CHANGELOG updated?
 - [x] Issue raised/referenced?

## Developers certificate of origin
- [x] I certify that this contribution is covered by the LGPLv2.1+ license as defined in our [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

----
📚 Documentation preview 📚: https://mdanalysis--4640.org.readthedocs.build/en/4640/

### CODE REVIEW

**Comment by IAlibay** [comment]
The test is inherently flawed because we know that even at a fixed seed the conformer generation will differ on different machines (it's caused trouble elsewhere). We should just test against on the fly generated coordinates.

File: testsuite/MDAnalysisTests/core/test_universe.py [hunk 1]
```diff
@@ -284,21 +280,19 @@ def test_rdkit_kwargs(self):
 
 
     def test_coordinates(self):
+        # manually create the molecule to compare
```
Comment (RMeli) [inline testsuite/MDAnalysisTests/core/test_universe.py]: I think it's worth to expand this comment to say why this is necessary, as explained in https://github.com/MDAnalysis/mdanalysis/pull/4640#issuecomment-2242695321.

Comment (IAlibay) [reply testsuite/MDAnalysisTests/core/test_universe.py]: Something like this?

Also this is a merge approval @RMeli ? you didn't green tick so it won't auto merge 😅

Comment (RMeli) [reply testsuite/MDAnalysisTests/core/test_universe.py]: Perfec, thanks!

**Review by RMeli — APPROVED** [review]

**Comment by RMeli** [comment]
Mmmh... Not sure why I enabled auto-merge here (I didn't realize)... The few failures in CI are the usual timeouts.

**Pull request merged**